### DISCUSSION TITLE
/articles/sharepoint-framework-optimize-spfx-bundles-ditch-ie11-support/

### DISCUSSION THREAD

**Discussion by vtns-comment-importer** [Voitanos/voitanos.io-comments#150] [Comments]
Optimize bundles by disabling IE11 support in SPFx projects
https://www.andrewconnell.com/articles/sharepoint-framework-optimize-spfx-bundles-ditch-ie11-support/

**Comment by vtns-comment-importer** [comment]
```header
Originally posted by Ofer Gal on 2021-08-27 12:44
```

I changed to "compilerOptions": { "target": "esnext", ...  
But all my constants with React.useState( show this error:  
Type '\[string, Dispatch\>\]' must have a '\[Symbol.iterator\]()' method that returns an iterator.ts(2488)

While I have your attention, (maybe) do you know why I have to reload the browser (Chrome or Edge) 30 times or more until I see my code when using [sharepoint.com/\_layouts/15/...](http://sharepoint.com/_layouts/15/workbench.aspx?)  
Any trick to make it quicker?

Thanks

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Andrew Connell on 2021-08-30 06:32
```

Context? SPFx version & more specifically, what's your TypeScript & Rush stack compiler version (+target TypeScript version). Also need to see more of your code that's throwing the error.

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Ofer Gal on 2021-08-30 09:10
```

SPFx 1.12.1  
"extends": "./node\_modules/@microsoft/rush-stack-compiler-3.7/includes/tsconfig-web.json"  
Using the out of the box SPFx TypeScript, react versions etc...  
All the state definitions using = React.useState(\[\]) are affected by it. (I use lots of them)  
The error is from typescript: "Type '\[any, Dispatch\]' must have a '\[Symbol.iterator\]()' method that returns an iterator.ts(2488)  
"  
![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/04f1dbf9ba7b49b1a59c604ab7bc213959ea44d007ae907f08296fbdd9ce7601.png)

All the esN but "target": "es5" cause the error.

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Andrew Connell on 2021-09-01 05:31
```

The spread operator was not known to the TSC when it imported type declarations & libraries. To address this, add \`es6\` to the \`lib\` array in your tsconfig.json.

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Ofer Gal on 2021-09-01 10:34
```

That did it! Thank you much!  
I even replaced "es5", for "es6",  
No errors!

**Comment by vtns-comment-importer** [comment]
```header
Originally posted by Markus on 2021-09-24 13:58
```

Thank you very much for all the information.  
I have created a new SPFx project and adjusted the parameters for "target" and "lib" in "tsconfig.json".  
During the build, an error appears regarding the "optional chaining". With "es2018" it works without problems.  
Is there a solution for this?  
![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/7e8fb870ae6d3e99d87dc5b64057a9c5b31e80ebf2ad780d3fad8a9a53cb9e47.png) ![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/3111bfe2198718ac9c629000c0acd458f816c3cd422f298b4a9cb1957e842036.png) ![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/84eaef4695267d5d28d78c3af0aea6323d7bb00a74011726f91d094f06b487ba.png)

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Andrew Connell on 2021-09-27 06:37
```

You can use whatever library is available within the "lib" section... in the blog post I just used the example of how ES6 is needed to support the spread operator. You're using optional chaining which is more recent. You just have to determine if your version of TypeScript supports it (it does) and which version of JS should support it.

**Comment by allenj3304** [comment]
Are you publishing you SPFx solutions to MS Marketplace? Certification 1160.2 states support IE11 is required (see https://learn.microsoft.com/en-us/legal/marketplace/certification-policies#11602-functionality). Found this researching the depreciation ES5 in TS 7.0

**Reply by andrewconnell** [reply]
That's for SharePoint add-ins, not SharePoint Framework. It's not relevant anyway... Microsoft 365 doesn't support IE11.
